dom/src/main/java/org/apache/james/mime4j/message/MessageBuilder.java

```diff
@@ -75,15 +75,68 @@
     private BodyFactory bodyFactory;
     private boolean flatMode;
     private boolean rawContent;
+    private final boolean ensureDateNotNull;
```

Comment:
@chibenwa : About your 2 tell me how much library we use in James and how much mailing list of those library do you follow, how much time you tell you "how about library X let's google and found blog post speaking about bug and breaking change on it". "WARNING log" is not a great solution neither because, it will be really easy for someone to miss a log when there are so many and moreover it is at runtime.

> Sorry, but I value API stability a lot, when it comes to things people are using.
If you do, and if you take time to think about my solution then you will agree with my solution. I come to this solution after thinking for a long time on how to fix the bug in a way that respect two conditions: API stabilility and user warning.

Also not that API stability is not all, a stable API which is ugly is for me not very usable and a `.notOverrideNullDate()` would be ugly IMO.

> Deprecating an API is a bit antagonist to API stability, isn't it?

No it's the contrary deprecating API is usefull when you do not want to break API !

> I'm against this deprecation.

I would be against the solution you are proposing too.
> When I see such changes, I just feel: "why should I bother, update my code in lots of places for just names changes?"

For my part, I don't when something is deprecated I know there is good reason behind it and I read the comment that explain the deprecation.